src/tribol/mesh/MfemData.hpp

```diff
@@ -810,6 +810,200 @@ class MfemMeshData
     return velocity_->GetRedecompFieldPtrs();
   }
 
+  /**
+   * @brief Clears all kinematic and rate penalty data
+   */
+  void ClearAllPenaltyData();
+
+  /**
+   * @brief Clears rate penalty data
+   */
+  void ClearRatePenaltyData();
+
+  /**
+   * @brief Sets the penalty parameter for the first registered Tribol mesh
+   * 
+   * @param penalty Penalty value for the first registered Tribol mesh
+   */
+  void SetMesh1Penalty(real penalty) { penalty_1_ = std::make_unique<double>(penalty); }
+
+  /**
+   * @brief Sets the penalty parameter for the second registered Tribol mesh
+   * 
+   * @param penalty Penalty value for the second registered Tribol mesh
+   */
+  void SetMesh2Penalty(real penalty) { penalty_2_ = std::make_unique<double>(penalty); }
+
+  /**
+   * @brief Get the penalty parameter for the first registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh1Penalty() const { return penalty_1_.get(); }
+
+  /**
+   * @brief Get the penalty parameter for the second registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh2Penalty() const { return penalty_2_.get(); }
+
+  /**
+   * @brief Sets the penalty scale for the first registered Tribol mesh
+   * 
+   * @param scale Penalty scale value for the first registered Tribol mesh
+   */
+  void SetMesh1PenaltyScale(real scale) { scale_1_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Sets the penalty scale for the second registered Tribol mesh
+   * 
+   * @param scale Penalty scale value for the second registered Tribol mesh
+   */
+  void SetMesh2PenaltyScale(real scale) { scale_2_ = std::make_unique<double>(scale); }
+
+  /**
+   * @brief Get the penalty scale for the first registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh1PenaltyScale() const { return scale_1_.get(); }
+
+  /**
+   * @brief Get the penalty scale for the second registered Tribol mesh
+   *
+   * @return const real* 
+   */
+  const real* GetMesh2PenaltyScale() const { return scale_2_.get(); }
+
+  /**
+   * @brief Sets the rate penalty for the first registered Tribol mesh
+   * 
+   * @param penalty Rate penalty value for the first registered Tribol mesh
+   */
+  void SetMesh1RatePenalty(real penalty) { rate_penalty_1_ = std::make_unique<double>(penalty); }
```

Comment:
Update 'Rate' getters/setters to reflect whether it is rate constant or rate percent/ratio